integration-testing/build.gradle

```diff
@@ -62,19 +74,28 @@ task debugAgentTest(type: Test) {
     classpath = sourceSet.runtimeClasspath
 }
 
+task coreAgentTest(type: Test) {
+    def sourceSet = sourceSets.coreAgentTest
+    dependsOn(project(':kotlinx-coroutines-core').jvmJar)
+    jvmArgs ('-javaagent:' + project(':kotlinx-coroutines-core').jvmJar.outputs.files.getFiles()[0])
+    testClassesDirs = sourceSet.output.classesDirs
+    classpath = sourceSet.runtimeClasspath
+}
+
 dependencies {
     testCompile "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
     testCompile 'junit:junit:4.12'
     npmTestCompile 'org.apache.commons:commons-compress:1.18'
     npmTestCompile 'com.google.code.gson:gson:2.8.5'
     debugAgentTestCompile project(':kotlinx-coroutines-core')
     debugAgentTestCompile project(':kotlinx-coroutines-debug')
+    coreAgentTestCompile project(':kotlinx-coroutines-core')
 }
 
 compileTestKotlin {
     kotlinOptions.jvmTarget = "1.8"
 }
 
 test {
-    dependsOn([npmTest, mavenTest, debugAgentTest])
+    dependsOn([npmTest, mavenTest, debugAgentTest, coreAgentTest])
```

Comment:
It has just occurred to me that we are wrong in adding all the dependencies to the `test` task as it makes it impossible to ask for `gradlew test` meaning "run all the regular tests only". We should be adding them all to the `check` task.